hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/multipart/S3MultipartUploadAbortRequest.java

```diff
@@ -116,36 +122,51 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
         throw new OMException("Abort Multipart Upload Failed: volume: " +
             volumeName + "bucket: " + bucketName + "key: " + keyName,
             OMException.ResultCodes.NO_SUCH_MULTIPART_UPLOAD_ERROR);
-      } else {
-        multipartKeyInfo = omMetadataManager
-            .getMultipartInfoTable().get(multipartKey);
-
-        multipartKeyInfo.setUpdateID(transactionLogIndex);
-
-        // Update cache of openKeyTable and multipartInfo table.
-        // No need to add the cache entries to delete table, as the entries
-        // in delete table are not used by any read/write operations.
-        omMetadataManager.getOpenKeyTable().addCacheEntry(
-            new CacheKey<>(multipartKey),
-            new CacheValue<>(Optional.absent(), transactionLogIndex));
-        omMetadataManager.getMultipartInfoTable().addCacheEntry(
-            new CacheKey<>(multipartKey),
-            new CacheValue<>(Optional.absent(), transactionLogIndex));
       }
 
-      omClientResponse = new S3MultipartUploadAbortResponse(multipartKey,
-          multipartKeyInfo,
+      // Check the OpenKeyTable if this transaction is a replay of ratis logs.
+      if (isReplay(ozoneManager, omKeyInfo.getUpdateID(), trxnLogIndex)) {
```

Comment:
How this check will help here, as we don't set updateID for open key OmKeyInfo.

Example:
 T1 -> MPU Initiate
T2 -> MPU abort

Now replay, T2 will not be considered as a replay. As OmKeyInfo will have updateID as 1.

Reply:
If the openKey was created after this transaction, then it is safe to assume that this is a replay.

Reply:
I see what you are saying now. MultipartUploadID would be unique even for the same keys for different initiates. Is that correct? If yes, ya we can remove this check here.